odoo/tests/common.py

```diff
@@ -2385,16 +2394,21 @@ def clear(self):
 
 def record_to_values(fields, record):
     r = {}
-    for f, descr in fields.items():
-        v = record[f]
+    # don't read the id explicitly, not sure why but if any of the "magic" hr
+    # field is read alongside `id` then it blows up e.g.
+    # james.read(['barcode']) works fine but james.read(['id', 'barcode'])
+    # triggers an ACL error on barcode, likewise km_home_work or
+    # emergency_contact or whatever. Since we always get the id anyway, just
+    # remove it from the fields to read
```

Comment:
@tivisse @rco-odoo I have no idea why it occurs but it's 100% reliable: go to `test_readonly_fields` and somewhere before the creation of the form add `james.read(['barcode'])` and `james.read(['id', 'barcode'])`, the first one will work with no issue, the second one will break every time claiming there's a problem reading `barcode`.